firmware/examples/Dac/dac.hpp

```diff
@@ -0,0 +1,99 @@
+    {
+        kBiasHigh  = 0,
+        kBiasLow   = 1
+    };
+    virtual void Initialize(void)                   = 0;
+    virtual bool WriteDac(uint16_t dac_output)      = 0;
+    virtual bool SetVoltage(float voltage)          = 0;
+    virtual void SetBias(BiasLevel bias_level)      = 0;
+};
+class Dac : public DacInterface
+{
+ public:
+    static LPC_DAC_TypeDef * dac_register;
+    static LPC_SC_TypeDef * sc_ptr;
+    static LPC_IOCON_TypeDef * iocon_register;
+    static constexpr float    kVref = 3.3f;
+    static constexpr uint8_t  kDacMode = 0b010;
+    static constexpr uint16_t kClearDac = 0b1111111111;
+    static constexpr uint8_t  kClockDiv = 0b0010;
+    enum Bit : uint8_t
+    {
+        kClockDividerReg = 1,
+        kBiasReg         = 16,
+        kDacOutReg       = 6,
+        kEnableDacReg    = 16,
+        kPullUpDown      = 3,
+        kFunc            = 0,
+        kPort0           = 0,
+        kPin26           = 26
+    };
+     constexpr Dac() :
+                        dac_(&dac_pin_),
+                        dac_pin_(kPort0, kPin26)
+     {
+     }
+     // For unit testing mocking purposes
+     explicit constexpr Dac(PinConfigureInterface * dac_pin) :
+                      dac_(dac_pin),
+                      dac_pin_(PinConfigure::CreateInactivePin())   // P0.26
+     {
+     }
+    // Initialize Dac by setting the clock divider and enabling
+    // The Dac Pin withiin IOCON.  Then set the initial Bias level to 0
+    void Initialize(void) override
+    {
+        dac_->SetPinFunction(kDacMode);
+        dac_->EnableDac(true);
+        dac_->SetAsAnalogMode(true);
+        dac_->SetPinMode(PinConfigureInterface::kInactive);
+        // Set Update Rate to 1MHz
+        dac_register->CR &= ~(1 << kBiasReg);
+    }
+    // The Dac ouptut is a 10 bit input and thus it is necessary to
+    // ensure dac_output is less than 1024(largest 10-bit number)
+    bool WriteDac(uint16_t dac_output) override
+    {
+        SJ2_ASSERT_FATAL(dac_output < 1023,
+                         "dac_output is above SJ2 output capacity."
+                         "output capacity must be between 0-1023.");
+        dac_register->CR &= ~(kClearDac << kDacOutReg);
+        dac_register->CR |= (dac_output << kDacOutReg);
+        return true;
+    }
+    // Takes an input voltage and converts the float value and calculates
+    // the conversion necessary and then typecasts it to an integer.
+    // If the voltage value is greater than 3.3 it will fail and end.
+    bool SetVoltage(float voltage) override
+    {
+        // value = (dac_out * 1024)/VrefP
+        float value = (voltage * 1024) / kVref;
+        int   conversion = static_cast<int>(value);
+        SJ2_ASSERT_FATAL(voltage < kVref,
+                         "Input voltage above SJ2"
+                         "output capacity of 3.3 Volts.");
+        dac_register->CR &= ~(kClearDac << kDacOutReg);
+        dac_register->CR |= (conversion << kDacOutReg);
```

Comment:
Since you have a `WriteDac` function, you should use it here. You want to minimize code duplication and maximize code reuse, everywhere you can.